src/xinterpreter.cpp

```diff
@@ -210,7 +213,19 @@ nl::json interpreter::execute_request_impl(int /*execution_counter*/,
       errorlevel = 1;
       // send the errors directly to std::cerr
       ename = "";
-      std::cerr << error_stream.str();
+      try {
```

Comment:
Can you add a FIXME saying that we should investigate the feature in clang that produces these codes automatically and just return them here instead of doing the manual work?